pkg/cidata/user-data.TEMPLATE

```diff
@@ -18,6 +18,129 @@ users:
     {{- end}}
 
 write_files:
```

Comment:
The `user-data.TEMPLATE` is becoming too long.
Eventually we should just place the scripts as plain files in the `cidata` drive.
(And we can execute scripts from the `cidata` drive directly, without copying them into `/var/lib/lima-guestagent`.)

Can be another PR, though.

Reply:
And in future we should also have shellcheck in CI

Reply:
shellcheck becomes much easier once the scripts are individually embedded templates; I just don't know if it will choke on the template expressions, like `{{.User}}` in the scripts.

There is also potentially a size limit on `user-data`, but I think the limit comes from the cloud-provider side, and not from cloud-init, so it probably doesn't apply to lima. But splitting out the files makes sense anyways.

Reply:
> Can be another PR, though.

Yes, this PR is complex enough. Let me know if you expect me to do it; or if you will do it as part of your regular refactoring...

Reply:
I'll do